docs/001_develop/03_client-capabilities/010_filters/foundation-filters.mdx

```diff
@@ -0,0 +1,437 @@
+      styles: css`
+        .container {
+          display: flex;
+          flex-direction: column;
+          gap: calc(var(--design-unit) * 3px);
+          padding: calc(var(--design-unit) * 4px);
+        }
+      `,
+      template: html`
+        <div class="container">
+          <rapid-text-field type="datetime-local" :value=${sync((x) => x.startDate)}>
+            Start Period
+          </rapid-text-field>
+          <rapid-text-field type="datetime-local" :value=${sync((x) => x.endDate)}>
+            End Period
+          </rapid-text-field>
+          <rapid-button @click=${(x) => x.checkWindow()}>Check</rapid-button>
+          <span>${(x) => x.resultString}</span>
+        </div>
+      `,
+    })
+    export class MyElement extends GenesisElement {
+      @observable startDate: string;
+      @observable endDate: string;
+      @observable resultString: string;
+      checkWindow() {
+        this.resultString = timeWindowFilter(this.startDate, this.endDate)
+          ? 'Exists within the start and end period'
+          : 'Doesnot fall between start and end period';
+      }
+    }
+
+    ```
+
+### With dependency injection(DI)
+
+    ```typescript
+    import { TimeWindow } from '@genesislcap/foundation-filters';
+
+    @customElement({
+      name: 'my-element',
+      styles: css`
+        .container {
+          display: flex;
+          flex-direction: column;
+          gap: calc(var(--design-unit) * 3px);
+          padding: calc(var(--design-unit) * 4px);
+        }
+      `,
+      template: html`
+        <div class="container">
+          <rapid-text-field type="datetime-local" :value=${sync((x) => x.startDate)}>
+            Start Period
+          </rapid-text-field>
+          <rapid-text-field type="datetime-local" :value=${sync((x) => x.endDate)}>
+            End Period
+          </rapid-text-field>
+          <rapid-button @click=${(x) => x.checkWindow()}>Check</rapid-button>
+          <span>${(x) => x.resultString}</span>
+        </div>
+      `,
+    })
+    export class MyElement extends GenesisElement {
+      @observable startDate: string;
+      @observable endDate: string;
+      @observable resultString: string;
+      @TimeWindow timeWindow: TimeWindow;
+      checkWindow() {
+        this.resultString = this.timeWindow.filter({start: this.startDate, end: this.endDate})
+          ? 'Exists within the start and end period'
+          : 'Doesnot fall between start and end period';
+      }
+    }
+
+    ```
+
+The function version is pretty self-explanatory, and you may be wondering why you would use the DI injected version. The
+reason is at the point of calling a filter, either directly or indirectly as (say) part of a chain, you may not know what
+data or state from the application it needs to provide its outcome. The DI versions encapsulate this complexity, which aids later refactoring, and leaves callers free to pass only primitive input parameters where possible.
+
+### DI use cases
+
```

Comment:
@patrickoneill-genesis maybe you can link to your react and angular pages on this when you write them